src/Slot.ts

```diff
@@ -191,13 +203,44 @@ export function connectSlot<T = void, T2 = void>(
         ) => {
             const paramHandlers = handlers[transportKey][param] || []
             const handlerIndex = paramHandlers.indexOf(handler)
-            if (handlerIndex > -1) handlers[transportKey][param].splice(handlerIndex, 1)
-            if (getParamHandlers(param, handlers).length === 0) callLazyDisonnectCallbacks(param)
-            awaitHandlerRegistration(String(transportKey), param)
+
+            if (handlerIndex > -1)
+                handlers[transportKey][param].splice(handlerIndex, 1)
+
+            if (getParamHandlers(param, handlers).length === 0)
+                callLazyDisonnectCallbacks(param)
+
+            if (remoteHandlersConnected[transportKey])
+                awaitHandlerRegistration(String(transportKey), param)
+
         }
 
-        transport.addRemoteHandlerRegistrationCallback(slotName, remoteHandlerRegistered)
-        transport.addRemoteHandlerUnregistrationCallback(slotName, remoteHandlerUnregistered)
+        const remoteEventListChangedListener = () => {
```

Comment:
Naming `remoteEventListChangedListener` is a bit confusing along with `addRemoteEventListDeclarationListener`. Can we use naming more aligned with the two methods above so at least were consistent here.